cirq-core/cirq/testing/consistent_act_on.py

```diff
@@ -79,7 +79,7 @@ def assert_all_implemented_act_on_effects_match_unitary(
     __tracebackhide__ = True
     # pylint: enable=unused-variable
 
-    num_qubits_val = protocols.num_qubits(val)
+    num_qubits_val = protocols.num_qubits(val) or 1
```

Comment:
This is a red flag for me - do we really expect a zero-qubit `ActOnArgs` to behave identically to a 1-qubit `ActOnArgs` in all circumstances?

Reply:
It was at least a yellow flag for me. Zero qubits is not supported well in this design because it ends up being just an empty dictionary. I'll see what I can do to make it more robust. Maybe your suggestions above regarding the merge method will help.

Reply:
Fixed